### PR TITLE
Code reorganization to facilitate planned `Added/Changed` extensions

### PR DESCRIPTION
**Pull request by bzm3r**
`lib.rs` is becoming quite unwieldy with the new extensions planned to `Added/Changed`: for instance, there will be new `WorldQuery` adapters, `ReadTraits` iterators, and new `WriteTraits` iterators. In order to make the file easier to navigate, and thereby mitigate the risk of errors, I think it will be useful to reorganize the code

### CODE REVIEW

**Review by JoJoJet — CHANGES_REQUESTED** [review]
These files are way too small. I don't see much benefit in separating `multiple_read` and `multiple_write` from `all` -- same with `singular_*` and `one`.

**Comment by bzm3r** [comment]
@JoJoJet ah, `one` and `all` will be the files that will grow the most in the upcoming changes. `multiple_read` and `multiple_write` will also see significant growth. 

What I can do is keep this PR open, and work on the added/changed stuff. Once that PR is up, you'll have more information as to how things ought to be organized?

**Comment by bzm3r** [comment]
Okay, because it seemed that I can keep the `singular_*` files small even after change detection, I've collapsed them into `one`

**Comment by JoJoJet** [comment]
I will leave the decision up to you, but IMO it's fine to just merge those files. `all.rs` will be quite long, but that's okay.

File: src/lib.rs [hunk 1]
```diff
@@ -164,7 +164,7 @@
 //! ```
 //!
 //! Alternatively, if you expect to only have component implementing the trait for each entity,
-//! you can use the filter [`One`](crate::One). This has significantly better performance than iterating
+//! you can use the filter [`One`](One). This has significantly better performance than iterating
```
Comment (JoJoJet) [inline src/lib.rs:167]:
```suggestion
//! you can use the filter [`One`](crate::One). This has significantly better performance than iterating
```

Due to a limitation with the `cargo-rdme` tool, all doc links in the README must start with `crate::`.

File: src/lib.rs [hunk 2]
```diff
@@ -189,6 +189,21 @@
...
+//!     // ...
+//! ) {
+//!     for tooltip in &tooltips {
+//!         println!("changed tool tips: {}", tooltip.tooltip());
+//!     }
+//! }
+//! ```
+//!
+//! Similarly, there exist `ChangedOne`, `AddedOne`, and `AddedAll`.
+//!
```
Comment (JoJoJet) [inline src/lib.rs:206]: I don't think this belongs here?

**Review by JoJoJet — CHANGES_REQUESTED** [review]

**Comment by JoJoJet** [comment]
So, apparently cargo-rdme doesn't like re-exports, so that doc comment on line 167 of lib.rs needs to be `crate::one::One`

**Comment by JoJoJet** [comment]
Github won't let me push to this branch, so I'm just going to go ahead and merge this, then apply my nitpicks in a follow-up PR.

**Review by JoJoJet — APPROVED** [review]

**Pull request merged**